src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -464,6 +464,104 @@ public void It_doesnt_includes_requires_preview_features()
             Assert.False(contains);
         }
 
+        [Theory]
+        [InlineData(true, true, "net6.0", false)]
+        [InlineData(true, true, "net7.0", true)]
+        [InlineData(true, true, ToolsetInfo.CurrentTargetFramework, true)]
+        [InlineData(true, false, ToolsetInfo.CurrentTargetFramework, true)]
+        [InlineData(false, false, ToolsetInfo.CurrentTargetFramework, true)]
+        public void TestDisableRuntimeMarshalling(bool disableRuntimeMarshalling, bool generateDisableRuntimeMarshallingAttribute, string targetFramework, bool shouldHaveAttribute)
+        {
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld", identifier: $"{disableRuntimeMarshalling}${generateDisableRuntimeMarshallingAttribute}${targetFramework}")
```

Comment:
I'd suggest using the `TestProject` class here instead of `CopyTestAsset`.  That lets you set various properties for the project to be created, which means you usually don't have to manipulate the XML with `WithProjectChanges`.  Here's an example of how this is done: https://github.com/dotnet/sdk/blob/f051b536cc12190488231f3a889df44214c1bc2e/src/Tests/Microsoft.NET.Build.Tests/AppHostTests.cs#L354-L365